cmd/uplink/cmd/ls.go

```diff
@@ -68,7 +83,18 @@ func list(cmd *cobra.Command, args []string) error {
 					if err != nil {
 						return err
 					}
-					err = listFiles(ctx, metainfo, prefix, true)
+
+					bucket, err := project.OpenBucket(ctx, bucket.Name, &access)
+					if err != nil {
+						return err
+					}
+
+					err = listFiles(ctx, bucket, prefix, true)
+					if err != nil {
+						return err
+					}
+
+					err = bucket.Close()
```

Comment:
bucket won't be closed when error occur with `listFiles`. Maybe additional function `listFilesFromBucket` or something like that?